php/class-connect.php

```diff
@@ -870,6 +919,92 @@ public function switch_account() {
 		return $return;
 	}
 
+	/**
+	 * Check the REST API connectivity for Cloudinary's endpoints.
+	 *
+	 * @return array
+	 */
+	public static function check_rest_api_connectivity() {
+
+		$connectivity = self::test_rest_api_connectivity();
+		$plugin       = get_plugin_instance();
+		$notices      = get_option( $plugin::KEYS['notices'], array() );
+		$now          = current_time( 'timestamp' ); // phpcs:ignore WordPress.DateTime.CurrentTimeTimestamp.Requested
+
+		wp_clear_scheduled_hook( 'cloudinary_rest_api_connectivity' );
+
+		if ( $connectivity['working'] ) {
+			unset( $notices[ self::META_KEYS['notices'] ] );
+			if ( empty( $notices ) ) {
+				delete_option( $plugin::KEYS['notices'] );
+			} else {
+				update_option( $plugin::KEYS['notices'], $notices, false );
+			}
+
+			wp_schedule_single_event( $now + ( DAY_IN_SECONDS ), 'cloudinary_rest_api_connectivity' );
```

Comment:
@pereirinha - Perhaps just set a `$next_run` value, that gets set here and below, then schedule the event in a single place, before returning.